mrp_procurements_plan/models/procurement.py

```diff
@@ -64,5 +53,99 @@ def run(self, autocommit=False):
 class ProcurementPlan(models.Model):
     _inherit = 'procurement.plan'
 
+    product_id = fields.Many2one(
+        'product.product', string='Final Product')
+    qty_to_produce = fields.Integer(
+        string='Quantity to be produced')
+    mrp_bom_id = fields.Many2one(
+        'mrp.bom', string='MRP BoM')
     production_ids = fields.One2many(
         'mrp.production', 'plan', string='Productions', readonly=True)
+
+    @api.one
+    def action_import(self):
+        if self.product_id:
+            raise exceptions.Warning(_('Error!: You can not import'
+                                       ' procurements, there is defined an'
+                                       ' final product'))
+        super(ProcurementPlan, self).action_import()
+
+    @api.multi
+    def button_generate_procurements(self):
+        for plan in self:
+            if not plan.product_id:
+                raise exceptions.Warning(_('Error!: You must enter the final'
+                                           ' product'))
+            if not plan.qty_to_produce:
+                raise exceptions.Warning(_('Error!: You must enter the'
+                                           ' quantity to produce'))
+            if not plan.mrp_bom_id:
+                raise exceptions.Warning(_('Error!: No BoM found'))
+            if len(plan.procurement_ids) > 0 and self.state != 'cancel':
+                raise exceptions.Warning(_('Error!: Already generated'
+                                           ' procurements orders'))
+            plan._deploy_bom_details(plan.mrp_bom_id)
+        return True
+
+    def _deploy_bom_details(self, bom):
+        for line in bom.bom_line_ids:
+            self._create_procurement_from_bom_line(1, line.product_id,
+                                                   line.product_qty)
+            if line.child_line_ids:
+                self._calculate_bom_line_details(line.child_line_ids, 1)
+        return True
+
+    def _calculate_bom_line_details(self, child_line_ids, level):
```

Comment:
Esto lo siento mucho... pero no entiendo como hacerlo. Si lo hago todo en un método, ya parte de este método que no debería de ejecutar, que es cuando se llama a la funcion _create_procurement_from_bom_line por cada linea de la LdM. He visto un ejemplo por ahí, y lo único que se me ocurre, quedaría así:
def _deploy_bom_details(self, bom):
        for line in bom.bom_line_ids:
            self._create_procurement_from_bom_line(1, line.product_id,
                                                   line.product_qty)
            if line.child_line_ids:
                def _calculate_bom_line_details(self, child_line_ids, level):
                    level += 1
                    for line in child_line_ids:
                        self._create_procurement_from_bom_line(level, line.product_id,
                                                                                             line.product_qty)
                        if line.child_line_ids:
                            self._calculate_bom_line_details(line.child_line_ids, level)
                    return True

¿Te refieres a dejarlo así?